### PR TITLE
HPX backend fixes

### PR DESCRIPTION
**Pull request by msimberg**
Fixes:

- A few tests with the asynchronous HPX backend. I've added a few fences for some of these tests but I expect their real place would be somewhere else.
- Warnings in the backend
- Missing header in the HPX backend (this is backwards compatible)

### CODE REVIEW

File: core/src/Kokkos_HostSpace.hpp [hunk 1]
```diff
@@ -306,7 +306,9 @@ namespace Impl {
 template <class ExecutionSpace>
 struct DeepCopy<HostSpace, HostSpace, ExecutionSpace> {
   DeepCopy(void* dst, const void* src, size_t n) {
+    Kokkos::fence();
     hostspace_parallel_deepcopy(dst, src, n);
+    Kokkos::fence();
```
Comment (msimberg) [inline core/src/Kokkos_HostSpace.hpp]: These are only required if `n` is below the threshold for serial copying.

**Comment by dalg24** [comment]
OK to test

**Comment by crtrott** [comment]
Retest this please

File: core/src/Kokkos_CopyViews.hpp [hunk 2]
```diff
@@ -3418,6 +3418,7 @@ create_mirror_view_and_copy(
         Impl::MirrorViewType<Space, T, P...>::is_same_memspace>::type* =
         nullptr) {
   (void)name;
+  typename Space::execution_space().fence();
```
Comment (dalg24) [inline core/src/Kokkos_CopyViews.hpp]: cc @crtrott 
if we agree we want to fence here, there will be other overloads that need fences.

Comment (dalg24) [reply core/src/Kokkos_CopyViews.hpp]: Also this belongs to another separate pull request.

**Comment by dalg24** [comment]
@msimberg you get my approval if you drop 0f2eefb

**Comment by crtrott** [comment]
Actually those fences are correct ... The problem is that most of our host backends are synchronous, so some of our API implementation is missing stuff. I have it in my plan to fix that after this release (i.e. get rid of unnecessary fences in the specifically limited synchronoziation calls.

**Comment by msimberg** [comment]
> @msimberg you get my approval if you drop [0f2eefb](https://github.com/kokkos/kokkos/commit/0f2eefb364c7eefc911cd25f1f6fe5b40f529f21)

I'm with @crtrott on this ;) https://github.com/kokkos/kokkos/commit/0f2eefb364c7eefc911cd25f1f6fe5b40f529f21 might not have the fences in the right places, but fences have to be added *somewhere* for the asynchronous HPX backend to pass tests. I'm happy to remove them from this PR if the fences are going in soon anyway in another PR. Those tests failing are not really critical for us, other than having a clean CI for the backend.

**Comment by dalg24** [comment]
We will add the fences in a follow up PR.

**Comment by msimberg** [comment]
I've rebased and dropped the fence commit.

**Pull request merged**